ddtrace/contrib/internal/openai/utils.py

```diff
@@ -260,38 +268,45 @@ def _loop_handler(span, chunk, streamed_chunks):
     When handling a streamed chat/completion/responses,
     this function is called for each chunk in the streamed response.
     """
-
     if span.get_tag("openai.response.model") is None:
         if hasattr(chunk, "type") and chunk.type.startswith("response."):
             response = getattr(chunk, "response", None)
             model = getattr(response, "model", "")
         else:
             model = getattr(chunk, "model", "")
         span.set_tag_str("openai.response.model", model)
-    # Only run if the chunk is a completion/chat completion
+
+    response = getattr(chunk, "response", None)
+    if hasattr(chunk, "type") and chunk.type == "response.completed":
+        streamed_chunks[0].append(response)
+
+    # Completions/chat completions are returned as `choices`
     for choice in getattr(chunk, "choices", []):
         streamed_chunks[choice.index].append(choice)
     if getattr(chunk, "usage", None):
         streamed_chunks[0].insert(0, chunk)
 
 
-def _process_finished_stream(integration, span, kwargs, streamed_chunks, is_completion=False):
+def _process_finished_stream(integration, span, kwargs, streamed_chunks, operation_type=""):
     prompts = kwargs.get("prompt", None)
     request_messages = kwargs.get("messages", None)
     try:
-        if is_completion:
+        if operation_type == "response":
+            formatted_completions = streamed_chunks[0][0]
+        elif operation_type == "completion":
             formatted_completions = [
                 openai_construct_completion_from_streamed_chunks(choice) for choice in streamed_chunks
             ]
-        else:
+        elif operation_type == "chat":
             formatted_completions = [
                 openai_construct_message_from_streamed_chunks(choice) for choice in streamed_chunks
             ]
-        if integration.is_pc_sampled_span(span):
+        if integration.is_pc_sampled_span(span) and not operation_type == "response":
             _tag_streamed_response(integration, span, formatted_completions)
```

Comment:
Let's change the name of `_tag_streamed_response` to `_tag_streamed_completions` since it's confusing if this should also apply to responses.